### PR TITLE
Remove tensorflow comment from script using PyTorch

### PR DESCRIPTION
**Pull request by paplhjak**
Remove TensorFlow distribution class comment `tfp.distributions.Normal`, as the main scripts were moved to PyTorch.

### CODE REVIEW

**Comment by paplhjak** [comment]
I also find the following comment in the `train` method in `paac_continuous.py` a little confusing:

```
        # TODO: Train the actor using the sum of the following two losses:
        # - REINFORCE loss, i.e., the negative log likelihood of the `actions` in the
        #   `action_distribution` (using the `log_prob` method). You then need to sum
        #   the log probabilities of the action components in a single batch example.
        #   Finally, multiply the resulting vector by `(returns - predicted values)`
        #   and compute its mean. Note that the gradient must not flow through
        #   the predicted values (you can use `.detach()` if necessary).
```
Specifically `You then need to sum the log probabilities of the action components in a single batch example [...] compute its mean`.

Shouldn't we rather multiply the log probabilities by the advantage vector (returns - predicted values) elementwise and then either take a sum or a mean over the batch? (Modifying the learning rate accordingly) 

TLDR: SUM **and** then MEAN VS either SUM **or** MEAN

**Comment by foxik** [comment]
Thanks, a good catch! (I grep-ed `tf.`, but forgot about `tfp.` :sweat_smile:) Another community-work point is yours :+1:

> _Shouldn't we rather multiply the log probabilities_

The catch is that the result of `action_distribution.log_prob(actions)` is not a vector, but a matrix -- the `actions` is a vector of action components (as defined on https://ufal.mff.cuni.cz/~straka/courses/npfl139/2324/slides/?08#17), so the result of the `log_prob` has shape `[batch_size, num_action_components]`. So the instructions are to sum over the `num_action_components`, arriving at `[batch_size]`, which can then be multiplied by the `(returns - predicted values)`, which is also a vector. (Without the summation, multiplying the `log_prob` result with the mentioned vector would result in an error, because broadcasting aligns dimensions from the back.)

> _ sum or a mean over the batch?_

Note that the sum/mean is important only if the same optimizer optimizes also another loss (like the entropy penalty; if only a single loss is used [or if the losses do not share weights, like MSE on critic and REINFORCE on actor if they do not share weights], then Adam will normalize the gradients, making the difference between sum/mean nearly none).

**Comment by paplhjak** [comment]
> The catch is that the result of action_distribution.log_prob(actions) is not a vector, but a matrix -- the actions is a vector of action components (as defined on https://ufal.mff.cuni.cz/~straka/courses/npfl139/2324/slides/?08#17), so the result of the log_prob has shape [batch_size, num_action_components]. So the instructions are to sum over the num_action_components, arriving at [batch_size], which can then be multiplied by the (returns - predicted values), which is also a vector. (Without the summation, multiplying the log_prob result with the mentioned vector would result in an error, because broadcasting aligns dimensions from the back.)

I see, because the action space in `paac_continous.py` is just a single interval, the actions have a single component, so their shape is [batch, 1]. That can be multiplied with (returns - predicted) without any broadcasting errors and the final mean subsumes the sum (as it is a sum over a single component). 

Thanks.

**Comment by foxik** [comment]
> _I see, because the action space in paac_continous.py is just a single interval, the actions have a single component, so their shape is [batch, 1]. That can be multiplied with (returns - predicted) without any broadcasting errors_

But note that the multiplication computes incorrect result in that case. When you multiply a vector with shape `[batch, 1]` with a vector with shape `[batch]`, the result will be a matrix with shape `[batch, batch]`, which is not what you want. (Unless you add a dimension to `returns`, which is naturally just a vector `[batch]`; compared to `predicted`, which can easily be `[batch, 1]`, as it is computed by a dense layer with a single output.)

**Comment by paplhjak** [comment]
I am aware of this, thank you. :)

I indeed have shapes `[batch, 1]` for returns and predicted as well.
